File: CS/Q148684/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraGrid.Views.Base;

namespace Q148684 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            InitializePopupControls();
        }

        private List<bool> unboundData = new List<bool>();

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Orders' table. You can move, or remove it, as needed.
            this.ordersTableAdapter.Fill(this.nwindDataSet.Orders);
            order_DetailsTableAdapter1.Fill(nwindDataSet.Order_Details);
            InitializeUnboundData();
        }

        private SimpleDetailEditor simpleEditor;
        private GridDetailEditor gridEditor;
        private Size SimpleEditorSize = new Size(202, 136);
        private Size GridEditorSize = new Size(403, 218);

        private void InitializePopupControls() {
            simpleEditor = new SimpleDetailEditor();
            simpleEditor.DataSource = ordersBindingSource;
            simpleEditor.Dock = DockStyle.Fill;
            gridEditor = new GridDetailEditor();
            gridEditor.DataSource = ordersBindingSource;
            gridEditor.Dock = DockStyle.Fill;
            detailsEdit.PopupControl = new PopupContainerControl();
        }

        private void InitializeUnboundData() {
            for (int i = 0; i < nwindDataSet.Orders.Count; i++)
                unboundData.Add(i % 2 == 0);
        }

        private void OnDetailsEditQueryPopup(object sender, CancelEventArgs e) {
            PopupContainerEdit edit = (PopupContainerEdit)sender;
            if ((bool)gridView1.GetFocusedRowCellValue(colUseGrid)) {
                edit.Properties.PopupControl.Controls.Add(gridEditor);
                edit.Properties.PopupControl.Size = GridEditorSize;
            } else {
                edit.Properties.PopupControl.Controls.Add(simpleEditor);
                edit.Properties.PopupControl.Size = SimpleEditorSize;
            }
        }

        private void OnDetailsEditCloseUp(object sender, CloseUpEventArgs e) {
            ((PopupContainerEdit)sender).Properties.PopupControl.Controls.Clear();
        }

        private void OnGridViewCustomUnboundColumnData(object sender, CustomColumnDataEventArgs e) {
            if (e.Column == colUseGrid)
                if (e.IsGetData)
                    e.Value = unboundData[e.ListSourceRowIndex];
                else if (e.IsSetData)
                    unboundData[e.ListSourceRowIndex] = (bool)e.Value;
        }

        private void OnDetailsEditQueryDisplayText(object sender, QueryDisplayTextEventArgs e) {
            e.DisplayText = "Edit Details";
        }
    }
}

File: VB/Q148684/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraGrid.Views.Base

Namespace Q148684
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			InitializePopupControls()
		End Sub

		Private unboundData As List(Of Boolean) = New List(Of Boolean)()

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Orders' table. You can move, or remove it, as needed.
			Me.ordersTableAdapter.Fill(Me.nwindDataSet.Orders)
			order_DetailsTableAdapter1.Fill(nwindDataSet.Order_Details)
			InitializeUnboundData()
		End Sub

		Private simpleEditor As SimpleDetailEditor
		Private gridEditor As GridDetailEditor
		Private SimpleEditorSize As New Size(202, 136)
		Private GridEditorSize As New Size(403, 218)

		Private Sub InitializePopupControls()
			simpleEditor = New SimpleDetailEditor()
			simpleEditor.DataSource = ordersBindingSource
			simpleEditor.Dock = DockStyle.Fill
			gridEditor = New GridDetailEditor()
			gridEditor.DataSource = ordersBindingSource
			gridEditor.Dock = DockStyle.Fill
			detailsEdit.PopupControl = New PopupContainerControl()
		End Sub

		Private Sub InitializeUnboundData()
			For i As Integer = 0 To nwindDataSet.Orders.Count - 1
				unboundData.Add(i Mod 2 = 0)
			Next i
		End Sub

		Private Sub OnDetailsEditQueryPopup(ByVal sender As Object, ByVal e As CancelEventArgs) Handles detailsEdit.QueryPopUp
			Dim edit As PopupContainerEdit = CType(sender, PopupContainerEdit)
			If CBool(gridView1.GetFocusedRowCellValue(colUseGrid)) Then
				edit.Properties.PopupControl.Controls.Add(gridEditor)
				edit.Properties.PopupControl.Size = GridEditorSize
			Else
				edit.Properties.PopupControl.Controls.Add(simpleEditor)
				edit.Properties.PopupControl.Size = SimpleEditorSize
			End If
		End Sub

		Private Sub OnDetailsEditCloseUp(ByVal sender As Object, ByVal e As CloseUpEventArgs) Handles detailsEdit.CloseUp
			CType(sender, PopupContainerEdit).Properties.PopupControl.Controls.Clear()
		End Sub

		Private Sub OnGridViewCustomUnboundColumnData(ByVal sender As Object, ByVal e As CustomColumnDataEventArgs) Handles gridView1.CustomUnboundColumnData
			If e.Column Is colUseGrid Then
				If e.IsGetData Then
					e.Value = unboundData(e.ListSourceRowIndex)
				ElseIf e.IsSetData Then
					unboundData(e.ListSourceRowIndex) = CBool(e.Value)
				End If
			End If
		End Sub

		Private Sub OnDetailsEditQueryDisplayText(ByVal sender As Object, ByVal e As QueryDisplayTextEventArgs) Handles detailsEdit.QueryDisplayText
			e.DisplayText = "Edit Details"
		End Sub
	End Class
End Namespace